src/default-directory-provider.coffee

```diff
@@ -42,3 +42,15 @@ class DefaultDirectoryProvider
   # * `null` if the given URI is not compatibile with this provider.
   directoryForURI: (uri) ->
     Promise.resolve(@directoryForURISync(uri))
+
+  # Public: Normalizes path.
+  #
+  # * `uri` {String} The path that should be normalized.
+  #
+  # Returns a {String} with normalized path.
+  normalizePath: (uri) ->
+    # Normalize disk drive letter on Windows to avoid opening two buffers for the same file
+    pathWithNormalizedDiskDriveLetter = uri
+    if !!process.platform.match(/^win/) and matchData = uri.match(/^([A-Za-z]):/)
+      pathWithNormalizedDiskDriveLetter = "#{matchData[1].toUpperCase()}#{uri.slice(1)}"
```

Comment:
Why are you using `!!process.platform.match(/^win/)` and not just `process.platform.match(/^win/)`?

Also, you could potentially use the following construct, it seems cleaner to me:
```coffeescript
pathWithNormalizedDiskDriveLetter =
  if !!process.platform.match(/^win/) and matchData = uri.match(/^([A-Za-z]):/)
    pathWithNormalizedDiskDriveLetter = "#{matchData[1].toUpperCase()}#{uri.slice(1)}"
  else
    uri
```